examples/plotting/zoom_map_example.py

```diff
@@ -0,0 +1,67 @@
+"""
+=========================
+Zoomed-in cutout of a Map
+=========================
+
+This example demonstrates how to showcase a particular region of a full disk
+image within the same figure.
+
+This is a useful for figures that want to provide an overview of your subject.
+"""
+
+import matplotlib.pyplot as plt
+from matplotlib.patches import ConnectionPatch
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+import sunpy.map
+from sunpy.data.sample import AIA_171_IMAGE
+
+###############################################################################
+# We will start off by defining a region of interest that we want to focus on.
+# In this case, we want to focus on the center of the Sun.
+length = 200 * u.arcsec
+x0 = -50 * u.arcsec
+y0 = -30 * u.arcsec
+
+###############################################################################
+# Now we will create a Map and then crop the region of interest.
+smap = sunpy.map.Map(AIA_171_IMAGE)
+bottom_left = SkyCoord(x0 - length, y0-length,
+                       frame=smap.coordinate_frame)
+top_right = SkyCoord(x0 + length, y0+length,
+                     frame=smap.coordinate_frame)
+submap = smap.submap(bottom_left, top_right=top_right)
+
+fig = plt.figure(figsize=(16, 5))
+ax1 = fig.add_subplot(1, 2, 1, projection=smap)
+
+smap.plot(axes=ax1, clip_interval=(1, 99.9)*u.percent)
+
+# Draw a box on the image highlighting the region of interest.
+smap.draw_rectangle(bottom_left, color='k', height=length * 2, width=length * 2)
+
+# Create a second axis on the plot which will be used to create the zoomed-in image.
```

Comment:
I think we might want this to be a section heading?

Reply:
Actually I did them as per last review recommendation!

Reply:
Ah I see.